tracer/src/Datadog.Trace/Util/HexString.cs

```diff
@@ -0,0 +1,174 @@
+﻿// <copyright file="HexString.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Diagnostics.Contracts;
+using System.Runtime.CompilerServices;
+using Datadog.Trace.DataStreamsMonitoring.Utils;
+
+namespace Datadog.Trace.Util;
+
+/// <summary>
+/// Utility class with helpers method that wrap HexConverter to convert values to and from hexadecimal strings.
+/// </summary>
+internal static class HexString
+{
+#if !NETCOREAPP3_1_OR_GREATER
+    [ThreadStatic]
+    private static byte[]? _buffer8; // always 8 bytes for ulong
+
+    [ThreadStatic]
+    private static byte[]? _buffer1; // always 1 byte
+#endif
+
+    [Pure]
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static ulong ReverseIfLittleEndian(ulong value)
+    {
+        return BitConverter.IsLittleEndian ? BinaryPrimitivesHelper.ReverseEndianness(value) : value;
+    }
+
+#if NETCOREAPP3_1_OR_GREATER
+    public static void ToHexChars(ReadOnlySpan<byte> bytes, Span<char> chars, bool lowerCase = true)
+#else
+    public static void ToHexChars(byte[] bytes, char[] chars, bool lowerCase = true)
```

Comment:
I don't see any tests for this one? I think `HexString` feels like something we should be covering 100% in unit tests 🙂

Reply:
Good catch, thanks. I added many new tests and discovered a few issues in `HexString` which I fixed as well.